cpp/src/io/utilities/column_buffer.cpp

```diff
@@ -149,14 +149,17 @@ std::unique_ptr<column> make_column(column_buffer_base<string_policy>& buffer,
                                     std::optional<reader_column_schema> const& schema,
                                     rmm::cuda_stream_view stream)
 {
-  if (schema_info != nullptr) { schema_info->name = buffer.name; }
+  if (schema_info != nullptr) {
+    schema_info->name        = buffer.name;
+    schema_info->is_nullable = buffer.is_nullable;
+  }
 
   switch (buffer.type.id()) {
     case type_id::STRING:
       if (schema.value_or(reader_column_schema{}).is_enabled_convert_binary_to_strings()) {
         if (schema_info != nullptr) {
-          schema_info->children.push_back(column_name_info{"offsets"});
-          schema_info->children.push_back(column_name_info{"chars"});
+          schema_info->children.push_back(column_name_info{"offsets", false});
```

Comment:
default value in the ctor would clean up these uses (see other comment).

Reply:
I think we want these specified explicitly. At least in the past these extra children have been non-nullable. Could be convinced otherwise.

Reply:
ok, removed the nullabilty on cudf-only bits...had to change the test but output schema still seems fine